spec.html

```diff
@@ -36049,6 +36069,117 @@ <h1>
+            1. Return _eUTF_.
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-match-records">
+          <h1>Match Records</h1>
+          <p>A <dfn>Match Record</dfn> is a Record value used to encapsulate the start and end indices of a regular expression match or capture.</p>
+          <p>Match Records have the fields listed in <emu-xref href="#table-match-record"></emu-xref>.</p>
+          <emu-table id="table-match-record" caption="Match Record Fields">
+            <table>
+              <tr>
+                <th>Field Name</th>
+                <th>Value</th>
+                <th>Meaning</th>
+              </tr>
+              <tr>
+                <td>[[StartIndex]]</td>
+                <td>An integer &ge; 0.</td>
+                <td>The number of code units from the start of a string at which the match begins (inclusive).</td>
+              </tr>
+              <tr>
+                <td>[[EndIndex]]</td>
+                <td>An integer &ge; [[StartIndex]].</td>
+                <td>The number of code units from the start of a string at which the match ends (exclusive).</td>
+              </tr>
+            </table>
+          </emu-table>
+        </emu-clause>
+
+        <emu-clause id="sec-getmatchstring" type="abstract operation">
+          <h1>
+            GetMatchString (
+              _S_: a String,
+              _match_: a Match Record,
+            )
+          </h1>
+          <dl class="header">
+          </dl>
+          <emu-alg>
+            1. Assert: _match_.[[StartIndex]] is an integer value &ge; 0 and &le; the length of _S_.
+            1. Assert: _match_.[[EndIndex]] is an integer value &ge; _match_.[[StartIndex]] and &le; the length of _S_.
+            1. Return the substring of _S_ from _match_.[[StartIndex]] to _match_.[[EndIndex]].
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-getmatchindicesarray" type="abstract operation">
+          <h1>
+            GetMatchIndicesArray (
+              _S_: a String,
+              _match_: a Match Record,
+            )
+          </h1>
+          <dl class="header">
+          </dl>
+          <emu-alg>
+            1. Assert: _match_.[[StartIndex]] is an integer value &ge; 0 and &lt; the length of _S_.
+            1. Assert: _match_.[[EndIndex]] is an integer value &ge; _match_.[[StartIndex]] and &le; the length of _S_.
+            1. Return ! CreateArrayFromList(&laquo; 𝔽(_match_.[[StartIndex]]), 𝔽(_match_.[[EndIndex]]) &raquo;).
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-makeindicesarray" type="abstract operation">
+          <h1>
+            MakeIndicesArray (
+              _S_: a String,
+              _indices_: a List,
+              _groupNames_: a List or *undefined*,
```

Comment:
`_groupNames_` can't be `*undefined*`: step 3 asserts that it has elements.

```suggestion
              _groupNames_: a List, each of whose elements is a String or *undefined*,
```

Reply:
Nice catch, I accidentally overwrote that in the rebase.